Mathlib/LinearAlgebra/Matrix/SemiringInverse.lean

```diff
@@ -0,0 +1,275 @@
+/-
+Copyright (c) 2024 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import Mathlib.Algebra.Algebra.Defs
+import Mathlib.Algebra.CharZero.Lemmas
+import Mathlib.Data.Fintype.Perm
+import Mathlib.Data.Matrix.Mul
+import Mathlib.GroupTheory.MonoidLocalization.Basic
+import Mathlib.GroupTheory.Perm.Sign
+
+/-!
+# Nonsingular inverses over semirings
+
+This files proves `A * B = 1 ↔ B * A = 1` for square matrices over a commutative semiring.
+
+-/
+
+namespace Matrix
+
+open Equiv Finset
+
+variable {n m R : Type*} [Fintype m] [Fintype n] [DecidableEq m] [DecidableEq n] [CommSemiring R]
+
+variable (s : ℤˣ) (A B : Matrix n n R) (i j : n)
+
+/-- Filter by parity -/
+def filterp : Finset (Perm n) := univ.filter fun σ ↦ σ.sign = s
```

Comment:
This perhaps should be called `Perm.ofParity` or `Perm.withParity` or something, rather than being in the matrix namespace. I'm pretty sure I've needed this construction before in a determinant-like proof; maybe @urkud has seen it with his recent AlternatingMap work.